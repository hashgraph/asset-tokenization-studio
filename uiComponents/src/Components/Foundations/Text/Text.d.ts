import type { BoxProps as ChakraBoxProps, TextProps as ChakraTextProps } from "@chakra-ui/layout";
import type { ComponentWithAs } from "@chakra-ui/system";
export interface TextProps extends ChakraBoxProps {
    size?: ChakraTextProps["size"];
    variant?: ChakraTextProps["variant"];
}
export declare const Text: ComponentWithAs<"p", TextProps>;
