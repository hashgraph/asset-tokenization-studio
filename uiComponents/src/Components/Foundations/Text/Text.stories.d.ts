import React from "react";
declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLParagraphElement>, HTMLParagraphElement>, any, import("./Text").TextProps, import("@chakra-ui/system/dist/system.types").As>>;
export default meta;
export declare const Default: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLParagraphElement>, HTMLParagraphElement>, any, import("./Text").TextProps, import("@chakra-ui/system/dist/system.types").As>>;
