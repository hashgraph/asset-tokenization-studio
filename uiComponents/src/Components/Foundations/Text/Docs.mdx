import {
  Story,
  Meta,
  Title,
  Controls,
  Primary,
  Stories,
  ArgTypes,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as TextStories from "./Text.stories";

<Meta of={TextStories} />

<Title />
<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/ioBricks-Design-System?type=design&node-id=1503%3A20660&mode=design&t=KlerdyEMWSQWLAVX-1"
  target="\_blank"
>
  View figma design
</a>
<Primary />

## Props

<ArgTypes of={TextStories} />

## Stories

<Stories title="" />
