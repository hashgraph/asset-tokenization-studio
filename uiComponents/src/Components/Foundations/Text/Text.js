var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { Box as ChakraBox } from "@chakra-ui/layout";
import { forwardRef, useStyleConfig as useChakraStyleConfig, } from "@chakra-ui/system";
import React from "react";
export var Text = forwardRef(function (_a, ref) {
    var children = _a.children, _b = _a.as, as = _b === void 0 ? "p" : _b, size = _a.size, variant = _a.variant, textStyle = _a.textStyle, props = __rest(_a, ["children", "as", "size", "variant", "textStyle"]);
    var styles = useChakraStyleConfig("Text", { size: size, variant: variant });
    return (React.createElement(ChakraBox, __assign({ ref: ref, as: as, textStyle: textStyle }, props, (!textStyle && { __css: styles })), children));
});
