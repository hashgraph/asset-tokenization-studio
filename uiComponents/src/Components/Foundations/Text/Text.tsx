import type {
  BoxProps as ChakraBoxProps,
  TextProps as ChakraTextProps,
} from "@chakra-ui/layout";
import { Box as ChakraBox } from "@chakra-ui/layout";
import type { ComponentWithAs } from "@chakra-ui/system";
import {
  forwardRef,
  useStyleConfig as useChakraStyleConfig,
} from "@chakra-ui/system";
import React from "react";

export interface TextProps extends ChakraBoxProps {
  size?: ChakraTextProps["size"];
  variant?: ChakraTextProps["variant"];
}

export const Text: ComponentWithAs<"p", TextProps> = forwardRef<TextProps, "p">(
  (
    { children, as = "p", size, variant, textStyle, ...props }: TextProps,
    ref
  ) => {
    const styles = useChakraStyleConfig("Text", { size, variant });

    return (
      <ChakraBox
        ref={ref}
        as={as}
        textStyle={textStyle}
        {...props}
        {...(!textStyle && { __css: styles })}
      >
        {children}
      </ChakraBox>
    );
  }
);
