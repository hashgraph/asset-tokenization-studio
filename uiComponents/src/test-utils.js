import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
