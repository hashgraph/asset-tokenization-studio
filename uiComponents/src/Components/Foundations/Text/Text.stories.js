var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
import React from "react";
import { ThemeStoryWrapper } from "../../story-config";
import { Text } from "./Text";
import { Title, Subtitle } from "@storybook/addon-docs";
import { Box, Divider, Grid, GridItem, Link } from "@chakra-ui/react";
import { textStyles } from "@/Theme/textStyles";
var bodyTextStyles = Object.keys(textStyles).filter(function (textStyle) {
    return textStyle.startsWith("Body");
});
var elementTextStyles = Object.keys(textStyles).filter(function (textStyle) {
    return textStyle.startsWith("Elements");
});
var meta = {
    title: "Design System/Foundations/Text",
    component: Text,
    argTypes: {
        textStyle: {
            options: __spreadArray(__spreadArray([], bodyTextStyles, true), elementTextStyles, true),
            control: { type: "select" },
            description: "The textStyle that the text has. Must be defined in the theme in the theme key `textStyles` "
        },
        size: {
            description: "The size of the text. We recommend using `textStyle` instead. "
        },
        variant: {
            description: "The variant of the text. Must be defined in the theme inside `Text` component."
        }
    },
    parameters: {
        docs: {
            page: function () { return (React.createElement(ThemeStoryWrapper, null,
                React.createElement(React.Fragment, null,
                    React.createElement(Title, null, "Text styles"),
                    React.createElement(Box, { my: 10 },
                        React.createElement(Subtitle, null, "Body Text")),
                    React.createElement(Grid, { templateRows: "repeat(4, 1fr)", gap: 10, alignItems: "center", gridAutoFlow: "column", justifyItems: "center", bg: "neutral" }, bodyTextStyles.map(function (name) { return (React.createElement(GridItem, { key: name },
                        React.createElement(Text, { textStyle: name }, name))); })),
                    React.createElement(Divider, { my: 10 }),
                    React.createElement(Box, { mb: 10 },
                        React.createElement(Subtitle, null, "Elements")),
                    React.createElement(Grid, { templateRows: "repeat(5, 1fr)", gap: 10, alignItems: "center", gridAutoFlow: "column", justifyItems: "center", bg: "neutral", maxW: "2000px", overflowX: "scroll" }, elementTextStyles.map(function (name) { return (React.createElement(GridItem, { key: name },
                        React.createElement(Text, { textStyle: name }, name))); })),
                    React.createElement(Box, { mt: 10 },
                        React.createElement(Link, { href: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1503%3A20660&t=gryWgf3cMhWIZY1U-4", isExternal: true }, "View figma design"))))); }
        }
    },
    args: {}
};
export default meta;
var Template = function (args) { return (React.createElement(Text, __assign({}, args), "Text example")); };
export var Default = Template.bind({});
Default.args = {};
