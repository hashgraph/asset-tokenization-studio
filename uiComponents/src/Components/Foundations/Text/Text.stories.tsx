import type { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { ThemeStoryWrapper } from "../../story-config";
import { Text } from "./Text";
import { Title, Subtitle } from "@storybook/addon-docs";
import { Box, Divider, Grid, GridItem, Link } from "@chakra-ui/react";
import { textStyles } from "@/Theme/textStyles";

const bodyTextStyles = Object.keys(textStyles).filter((textStyle) =>
  textStyle.startsWith("Body")
);

const elementTextStyles = Object.keys(textStyles).filter((textStyle) =>
  textStyle.startsWith("Elements")
);

const meta = {
  title: "Design System/Foundations/Text",
  component: Text,
  argTypes: {
    textStyle: {
      options: [...bodyTextStyles, ...elementTextStyles],
      control: { type: "select" },
      description:
        "The textStyle that the text has. Must be defined in the theme in the theme key `textStyles` ",
    },
    size: {
      description:
        "The size of the text. We recommend using `textStyle` instead. ",
    },
    variant: {
      description:
        "The variant of the text. Must be defined in the theme inside `Text` component.",
    },
  },
  parameters: {
    docs: {
      page: () => (
        <ThemeStoryWrapper>
          <>
            <Title>Text styles</Title>

            <Box my={10}>
              <Subtitle>Body Text</Subtitle>
            </Box>

            <Grid
              templateRows="repeat(4, 1fr)"
              gap={10}
              alignItems="center"
              gridAutoFlow={"column"}
              justifyItems="center"
              bg="neutral"
            >
              {bodyTextStyles.map((name) => (
                <GridItem key={name}>
                  <Text textStyle={name}>{name}</Text>
                </GridItem>
              ))}
            </Grid>

            <Divider my={10} />
            <Box mb={10}>
              <Subtitle>Elements</Subtitle>
            </Box>

            <Grid
              templateRows="repeat(5, 1fr)"
              gap={10}
              alignItems="center"
              gridAutoFlow={"column"}
              justifyItems="center"
              bg="neutral"
              maxW={"2000px"}
              overflowX="scroll"
            >
              {elementTextStyles.map((name) => (
                <GridItem key={name}>
                  <Text textStyle={name}>{name}</Text>
                </GridItem>
              ))}
            </Grid>
            <Box mt={10}>
              <Link
                href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1503%3A20660&t=gryWgf3cMhWIZY1U-4"
                isExternal
              >
                View figma design
              </Link>
            </Box>
          </>
        </ThemeStoryWrapper>
      ),
    },
  },
  args: {},
} as Meta<typeof Text>;
export default meta;

const Template: StoryFn<typeof Text> = (args) => (
  <Text {...args}>Text example</Text>
);

export const Default = Template.bind({});
Default.args = {};
