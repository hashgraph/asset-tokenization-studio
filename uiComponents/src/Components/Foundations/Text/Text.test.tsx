import React from "react";
import { render } from "../../../test-utils";
import type { TextProps } from "./Text";
import { Text } from "./Text";

describe(`<Text />`, () => {
  const defaultProps = {
    fontSize: "32px",
  };

  const themeCustom = {
    textStyles: {
      red: {
        color: "red",
      },
    },
    components: {
      Text: {
        variants: {
          regular: {
            fontFamily: "Helvetica Neue",
          },
        },
      },
    },
  };

  const factoryComponent = (props: TextProps = defaultProps) =>
    render(
      <Text {...props} data-testid="Text">
        Text example
      </Text>,
      themeCustom
    );

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
  });

  test("renders correctly with text style", () => {
    const component = factoryComponent({ textStyle: "red" });

    expect(component.asFragment()).toMatchSnapshot("WithTextStyle");
  });

  test("renders correctly with variant", () => {
    const component = factoryComponent({ variant: "regular" });

    expect(component.asFragment()).toMatchSnapshot("WithVariant");
  });

  test("renders correctly with custom Text tag", () => {
    const component = factoryComponent({ as: "span" });

    expect(component.getByTestId("Text").tagName).toBe("SPAN");
    expect(component.asFragment()).toMatchSnapshot("WithCustomTag");
  });
});
