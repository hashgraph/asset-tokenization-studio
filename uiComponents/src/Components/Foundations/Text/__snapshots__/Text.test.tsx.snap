// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders correctly 1`] = `
<DocumentFragment>
  <p
    class="css-idm3bz"
    data-testid="Text"
  >
    Text example
  </p>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Text /> renders correctly with custom Text tag: WithCustomTag 1`] = `
<DocumentFragment>
  <span
    class="css-0"
    data-testid="Text"
  >
    Text example
  </span>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Text /> renders correctly with text style: WithTextStyle 1`] = `
<DocumentFragment>
  <p
    class="css-tokvmb"
    data-testid="Text"
  >
    Text example
  </p>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Text /> renders correctly with variant: WithVariant 1`] = `
<DocumentFragment>
  <p
    class="css-0"
    data-testid="Text"
  >
    Text example
  </p>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
